orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/DependentPipelineStarter.groovy

```diff
@@ -133,6 +136,25 @@ class DependentPipelineStarter implements ApplicationContextAware {
     // This is required for template source with jinja expressions
     trigger.artifacts = pipelineConfig.receivedArtifacts
 
+    // expectedArtifacts for triggers are defined at the top level of the
```

Comment:
Trying to follow this, I started above with:
```
    def expectedArtifactIds = pipelineConfig.get("triggers", []).findAll {
      it.type == "pipeline" && it.pipeline == parentPipeline.pipelineConfigId
    } collectMany {
      it.expectedArtifactIds ?: []
    }
```
I'd love some help connecting what that block does to here.